### PR TITLE
Add Refresh buttons on listings issue #12456

### LINKED ISSUE #12456: Front: refresh button
##### **Overview of the issue**

When viewing an entity or the user-management page, it would be nice to have a `refresh` button.

##### **Motivation for or Use Case**

No need to press F5 :-)

##### **Reproduce the error**

- generate an application (Angular, React or Vue)
- generate entities (not mandatory)
- start the app
- go to user-management
- or go to an entity

##### **Suggest a Fix**

- add refresh button in front page
- link the action to retrieve users or entities
- do it for all front: Angular, React, Vue

##### **Browsers and Operating System**

-   [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by FabienEssid**
Fixed #12456 

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management.component.html.ejs [hunk 1]
```diff
@@ -19,15 +19,22 @@
...
-        <button id="jh-create-entity" data-cy="entityCreateButton" class="btn btn-primary float-right jh-create-entity create-<%= entityUrl %>" [routerLink]="['/<%= entityUrl %>/new']">
-            <fa-icon icon="plus"></fa-icon>
-            <span <% if (searchEngine !== false) { %>class="hidden-sm-down" <% } %> jhiTranslate="<%= i18nKeyPrefix %>.home.createLabel">
-            Create a new <%= entityClassHumanized %>
-            </span>
-        </button>
-        <%_ } _%>
+
+        <div class="d-flex justify-content-end">
+          <button class="btn btn-primary mr-2" (click)="handleSyncList()" [disabled]="isLoading">
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management.component.html.ejs:24]: can you check the indentation here plz ? It should be with 4 spaces for Angular

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/pagination-template.ejs [hunk 2]
```diff
@@ -56,8 +58,14 @@
                 query: this.currentSearch,
                 size: this.itemsPerPage,
                 sort: this.sort()}).subscribe(
-                    (res: HttpResponse<I<%= entityAngularName %>[]>) => this.onSuccess(res.body, res.headers, pageToLoad, !dontNavigate),
-                    () => this.onError()
+                    (res: HttpResponse<I<%= entityAngularName %>[]>) => {
+                      this.isLoading = false;
```
Comment (pascalgrimaud) [inline generators/entity-client/templates/angular/src/main/webapp/app/entities/pagination-template.ejs:62]: same comment with indent

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management.component.html.ejs (hunk 1 above)
Comment (FabienEssid) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/entity-management.component.html.ejs:24]: Yes you're right. I'm checking !

File: generators/entity-client/templates/angular/src/main/webapp/app/entities/pagination-template.ejs (hunk 2 above)
Comment (FabienEssid) [reply generators/entity-client/templates/angular/src/main/webapp/app/entities/pagination-template.ejs]: Fixed !

**Comment by pascalgrimaud** [comment]
There are some errors in react builds. Can you have a look @FabienEssid plz ?

**Comment by FabienEssid** [comment]
@pascalgrimaud i fixed issues in React builds. But now, I have a problem with Vue build but I'm not sure to understand what the problem is? Can you give me some help ?

**Comment by pascalgrimaud** [comment]
no worry, @FabienEssid : just restart the Vue Cassandra build, hope it should be ok :)

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs [hunk 3]
```diff
@@ -69,6 +76,10 @@ export default class <%=jhiPrefixCapitalized%>UserManagementComponent extends mi
     }
   }
 
+  public handleSyncList(): void {
```
Comment (pascalgrimaud) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs:79]: @FabienEssid : you need to move this function just before line 63, as it is needed for Cassandra application.

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs (hunk 3 above)
Comment (FabienEssid) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs:79]: Oh you right ! I will fix it in few minutes

**Comment by pascalgrimaud** [comment]
good job @FabienEssid 
is this PR ready ? Can we remove draft mode ?

**Comment by FabienEssid** [comment]
@pascalgrimaud the PR is ready, you can review it 😄

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Thanks and congrats for your 1st contribution @FabienEssid !

**Pull request merged**